domain/ai/main.go

```diff
@@ -0,0 +1,98 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/marmelab/labyrinth/domain/internal/model"
+)
+
+func main() {
+	lastWinningWeigths := model.NewBestHintWeights()
+
+outer:
+	for round := 0; round < 100; round++ {
+		fmt.Printf("----------\n")
+		fmt.Printf("[Round #%d] Start\n", round)
+
+		board, _ := model.NewBoard(7, 4)
+
+		// We want to keep some weights for optimization
+		for _, player := range board.Players {
+			switch player.Color {
```

Comment:
I don't really understand the principle of switching over colors. You want the IA to have multiple algorithm? 
Also is colors always the same for the IA?

Reply:
This file is used for AI weight training, and this switch allows to keep some weights for the next round to optimize the training